lib/src/api/metrics/meter.dart

```diff
@@ -0,0 +1,13 @@
+import 'package:opentelemetry/api.dart';
+
+abstract class Meter {
+  /// Creates a new [Counter] metric. Generally, this kind of metric is used
+  /// when the value is a quantity, the sum is of primary interest, and the
+  /// event count and value distribution are not of primary interest.
+  ///
+  /// [name] The name of the [Counter] See https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument-naming-rule"
+  /// [options] Options needed for metric creation. Pattern borrowed from
+  /// opentelemetry-js. This is optional.
+  ///
+  Counter createCounter<T>(String name, {MetricOptions options});
```

Comment:
For the generic to work, I think this will need to be:
```suggestion
  Counter<T> createCounter<T>(String name, {MetricOptions options});
```